Assets/GameManager/TowerManager.cs

```diff
@@ -95,16 +95,40 @@ public class TowerManager : MonoBehaviour {
     if (GameStateManager.Instance.Nu < cost) { return false; }
     GameStateManager.Instance.Nu -= cost;
     AddTowerPrice(data.type, cost);
-    StartCoroutine(BuildTower(waypoint, data));
+    Tower tower = ConstructTower(waypoint, data);
+    StartCoroutine(DisableTower(tower, buildDelay, () => {
```

Comment:
Reformat this, and other, lambdas at the call site for readability as discussed offline.